tools/underwater-gps.py

```diff
@@ -6,17 +6,105 @@
+       + "06,"                                        # Number of Satellites
+       + "1.2,"                                       # HDOP
+       + "0,"                                         # MSL altitude
+       + "M,"                                         # MSL altitude unit (Meters)
+       + "0,"                                         # Geoid separation
+       + "M,"                                         # Geoid separation unit (Meters)
+       + "00,"                                        # Age of differential GPS data, N/A
+       + "0000"                                       # Age of differential GPS data, N/A
+       + "*67")                                       # Checksum
+
+# https://www.trimble.com/oem_receiverhelp/v4.44/en/NMEA-0183messages_RMC.html
+gprmc = ("$GPRMC,"                                  # Message ID
+       + "{hours:02d}{minutes:02d}{seconds:02.4f}," # UTC Time
+       + "A,"                                       # Status A=active or V=void
+       + "{lat:02.0f}{latmin:02.6f},"               # Latitude (degrees + minutes)
+       + "{latdir},"                                # Latitude direction (N/S)
+       + "{lon:03.0f}{lonmin:02.6},"                # Longitude (degrees + minutes)
+       + "{londir},"                                # Longitude direction (W/E)
+       + "0.0,"                                    # Speed over the ground in knots
+       + "{orientation:03.2f},"                     # Track angle in degrees
+       + "{date},"                                  # Date
+       + ","                                        # Magnetic variation in degrees
+       + ","                                        # Magnetic variation direction
+       + "A"                                        # A=autonomous, D=differential,
+                                                    # E=Estimated, N=not valid, S=Simulator.
+       + "*76")                                     # Checksum
+
+# https://www.trimble.com/oem_receiverhelp/v4.44/en/NMEA-0183messages_VTG.html
+gpvtg = ("$GPVTG,"                    # Message ID
+       + "{orientation:03.2f},"       # Track made good (degrees true)
+       + "T,"                         # T: track made good is relative to true north
+       + ","                          # Track made good (degrees magnetic)1
+       + "M,"                         # M: track made good is relative to magnetic north
+       + "0.0,"                      # Speed, in knots
+       + "N,"                         # N: speed is measured in knots
+       + "0.0,"                       # Speed over ground in kilometers/hour (kph)
+       + "K,"                         # K: speed over ground is measured in kph
+       + "A"                          # A=autonomous, D=differential,
+                                      # E=Estimated, N=not valid, S=Simulator.
+       + "*0C")                       # Checksum
+
+def fixChecksum(string):
```

Comment:
I prefer this to be calculateNmeaChecksum() returning an int, and format the message appropriately in the code that is calling this.